### PR TITLE
Add test project

### LINKED ISSUE #29: Add Test project
The project folder should probably be organized into `src/` and `test/` folders, with a test project added.

The test project should have code which calls each function. The calls do not necessarily need to be valid, or even run, they just need to be, at minimum, present in a source file. This may be sufficient to force the Linker to lookup the symbol of each called method, and ensure it is present. This would help catch missing exported names from the .asm file, and name mangling changes caused by method signature updates.

In particular, marking a method as `const`, or changing from a `class` to a `struct` has the potential to change the mangled name of a method.

Proposed structure:
```
README.md
makefile
OP2Internal.sln
src/
  OP2Internal.vcxproj
  OP2Internal.vcxproj.filters
  OP2Internal.h
  OP2Internal.asm
  PointTypes.h
  WinTypes.h
  Game/
  Network/
  ResourceManagement/
  UserInterface
test/
  Test.vcxproj
  DummyMain.cpp  // We may want to rename this later
```

### PR DESCRIPTION
**Pull request by Brett208**
Eliminated the x64 configuration from the test project as I think it doesn't apply. 

gtest is setup like in OP2Utility, so we cannot use gmock on MSVC. Also, cannot specially compile with C++17 project settings. However, some C++17 features could be available in experimental settings like filesystem.

Test suite runs after compilation locally without warning or error. I took best stab at appveyor settings, but not tested.

Closes #29

### CODE REVIEW

**Comment by Brett208** [comment]
Looks like Git (or GitHub?) recognized the new file locations when moving to the src directory, but still counted every line as changed. I'm guessing white space issue again, although haven't invested that hunch yet. 👎

File: OP2Internal.asm [hunk 1]
```diff
@@ -12,7 +12,7 @@
 ; You must set the following to the load address of the DLL
 ; to ensure proper linking to the functions in Outpost2.exe
 ; *********************
-LoadAddr EQU 0x11000000
+LoadAddr EQU 0x14000000
```
Comment (DanRStevens) [inline OP2Internal.asm]: I assume this was committed by accident. This should not be changed. The default value for levels should be different from what NetFixClient uses.

File: .appveyor.yml [hunk 2]
```diff
@@ -4,9 +4,18 @@ configuration:
...
 install:
   - if not exist "%ProgramFiles%\NASM\"  choco install -y nasm
   - set PATH=%ProgramFiles%\NASM;%PATH%
   - nasm -v
+before_build: nuget restore
 build:
   project: OP2Internal.sln
+test_script:
+  - echo Current working directory = '%CD%'
+  - echo Test exe path = '%APPVEYOR_BUILD_FOLDER%\%CONFIGURATION%\Test.exe'
```
Comment (DanRStevens) [inline .appveyor.yml]: We can probably get rid of these two `echo` lines. They were used while debugging the AppVeyor config when first getting the test project running. They were never cleaned up afterwards.

**Review by DanRStevens — COMMENTED** [review]
Thank you very much for handling this.

I noticed the `LoadAddr` constant was updated from `0x11000000` to `0x14000000` in this change set. I think that part should be reverted.

----

I checked the large diff with:
`git show e4c0c37679906b68a5c069f0c09ce8f9489350bd --summary`
Alternatively it can be checked with:
`git show e4c0c37679906b68a5c069f0c09ce8f9489350bd --name-status`

Results show the files are between 95%-100% similar.

This seems to be an EOL conversion change. I was able to reverse most of the differences with:
`find . -name '*.h' -exec unix2dos {} \;`

It seems the older revision uses mostly DOS/Windows line endlings CRLF, while the newer revision uses mostly Linux LF line endings.

Of the remaining differences, I converted some the other direction with `dos2unix`, which eliminated many of those differences. Hence a few of the files were originally using Linux LF line endings.

In the end, only 3 files had real changes:
```
OP2Internal.asm   ; This change I think should be reverted
OP2Internal.sln   ; Change location of src project, and BOM change
test/Test.vcxproj ; Change location of src project reference
```

In summary, the renames look fine. It's just that one change to OP2Internal.asm which I think should be reverted before merging.

File: OP2Internal.asm (hunk 1 above)
Comment (Brett208) [reply OP2Internal.asm]: If I changed that, it was on accident. However it looks fine on my copy and in the repository. Not sure what is going on. Maybe you rewrote history?

**Comment by Brett208** [comment]
@DanRStevens you mentioned making some line ending changes, but I don't see the commits. If you want to do something to unify the line changes please go ahead. Otherwise, please merge when you are ready.

If you need me to make the line ending changes, just let me know what you want. Thanks.

File: OP2Internal.asm (hunk 1 above)
Comment (DanRStevens) [reply OP2Internal.asm]: I made no changes to history. My verification of changes was all done locally. As this is showing up on GitHub, that's where the change is being done.

**Comment by DanRStevens** [comment]
The line ending changes I did was just local verification of your changes. When you moved the files, it auto converted the line endings. I'm not sure why or how, but that's the commit that changed line endings.

I believe line endings should be normalized now, so no further changes needed.

At least not unless you want to edit history to separate out the line ending changes from the move. That might make for cleaner history, and easier to verify diffs, but isn't really needed. Do you want the changes separated out? Should be easy to do.

**Comment by DanRStevens** [comment]
I created a [AddTestProjectCleanup](https://github.com/OutpostUniverse/OP2Internal/tree/AddTestProjectCleanup) branch. It's a rewritten version of this branch. I took this branch, rebased it on master, (intended to edit history, though which also eliminated the merge of master), reverted the .asm file change, and added an extra commit to do the EOL conversion. This was added before the move commit, so all those files now show up as renames in the move commit.

Was pretty quick and easy to do. Adding the EOL conversion before the move commit naturally didn't create any conflicts, so the tools were able to handle it easily.

If you  like the changes, you could set the head pointer of this branch to the other branch, and force push here, or open a new pull request for the other branch. Or just revert that one .asm file change here, and merge as is.

To overwrite this branch with the rewritten version:
```
git checkout AddTestProject  ; Set current branch
git reset AddTestProjectCleanup  ; Reset HEAD to point to listed commit (HEAD of other branch)
git push -f  ; Force push
```

**Comment by Brett208** [comment]
Thanks for cleaning up. I'm happy just deleting this branch and merging the other.
